client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -58,30 +59,30 @@ public ReaderGroupConfigBuilder disableAutomaticCheckpoints() {
         * Add a stream and its associated start {@link StreamCut} and end {@link StreamCut} to be read by the
         * readers of a ReaderGroup.
         *
-        * @param scopedStreamName Scoped Name of the Stream.
-        * @param startStreamCut Start {@link StreamCut}
-        * @param endStreamCut End {@link StreamCut}
+        * @param scopedStreamName Scoped name of the stream, if only stream name is specified then {@link ReaderGroup#getScope()} is used for the scope.
+        * @param startStreamCut Start {@link StreamCut}.
+        * @param endStreamCut End {@link StreamCut}.
         * @return Reader group config builder.
         */
        public ReaderGroupConfigBuilder stream(final String scopedStreamName, final StreamCut startStreamCut, final StreamCut endStreamCut) {
-           final Stream stream = Stream.of(scopedStreamName);
 
            if (startingStreamCuts == null) {
                startingStreamCuts = new HashMap<>();
            }
-           this.startingStreamCuts.put(stream, startStreamCut);
+           this.startingStreamCuts.put(scopedStreamName, startStreamCut);
```

Comment:
This change results in a confusing state of affairs: "scopedStreamName" may or may not be scoped, and as such the stream names in `startingStreamCuts` are also maybe or maybe not scoped. We should find a cleaner way to do this.

Reply:
The problem is that there is some implicit behavior, which is prone to confusion and error. Perhaps one solution would be to have two constructors such that one takes both scope and stream names, while the second only takes the stream name only. The latter uses the RG scope as the default one. This way we do not make the behavior ambiguous. Would it work?

Reply:
I have updated the code such that the user can pass either scope and stream name or just the stream name.